WalletWasabi/Services/MemPoolService.cs

```diff
@@ -26,6 +26,32 @@ public MemPoolService()
 
 		private long _cleanupInProcess;
 
+		public void TryPerformMempoolCleanup(IEnumerable<uint256> allMempoolHashes)
+		{
+			if (Interlocked.Read(ref _cleanupInProcess) == 1) return; // If already cleaning, then no need to run it that often.
+			Interlocked.Exchange(ref _cleanupInProcess, 1);
```

Comment:
this code do not prevent two thread from running same time.

Reply:
It does and it is the most sure way to do it. 😄 

That being said, this code wouldn't be needed here at all I just left here for consistency with the old method, but that's another topic.

Reply:
No it does not.

1. T1: if (Interlocked.Read(ref _cleanupInProcess) == 1) // evaluated to false
2. T2: if (Interlocked.Read(ref _cleanupInProcess) == 1) // evaluated to false
3. T1: Interlocked.Exchange(ref _cleanupInProcess, 1);
4. T2: Interlocked.Exchange(ref _cleanupInProcess, 1);
5. Now `_cleanupInProcess == 2` and you execute the rest twice

Reply:
`Interlocked.Read(ref _cleanupInProcess)` just mean that you read the `_cleanupInProcess` atomically, nothing more. It does not guarantee that just after the OS make a context switch to another thread before the next incrementation.